### PR TITLE
Fix some named mutex tests

### PR DESCRIPTION
**Pull request by kouvel**
Related to dotnet/coreclr#5030
Related to dotnet/coreclr#3422

### CODE REVIEW

**Comment by kouvel** [comment]
@sergiy-k @stephentoub 

Flagged as [WIP] until the dependency change goes into coreclr

File: src/System.Threading/tests/MutexTests.cs [hunk 1]
```diff
@@ -147,27 +126,37 @@ public void OpenExisting_NameUsedByOtherSynchronizationPrimitive_Windows()
...
-                Assert.Equal(m, ame.Mutex);
+                    Task t = Task.Factory.StartNew(() =>
+                    {
+                        Assert.True(m.WaitOne(FailedWaitTimeout));
+                    }, CancellationToken.None, TaskCreationOptions.LongRunning, TaskScheduler.Default);
+                    t.Wait();
+                    AbandonedMutexException ame = Assert.Throws<AbandonedMutexException>(() => WaitHandle.WaitAny(new[] { m }, FailedWaitTimeout));
+                    Assert.Equal(0, ame.MutexIndex);
+                    Assert.Equal(m, ame.Mutex);
+                }
```
Comment (stephentoub) [inline src/System.Threading/tests/MutexTests.cs]: You might consider refactoring this a bit, e.g. (pseudo-code)
```C#
public static IEnumerable<object[]> AbandonExisting_MemberData()
{
    for (int waitType = 0; waitType < 2; waitType++) // 0 == Wait, 1 == WaitAny
    {
        yield return new object[] { null, waitType};
        yield return new object[] { Guid.NewGuid().ToString("N"), waitType };
    }
}

[Theory]
[MemberData(nameof(AbandonExisting_MemberData))]
public void AbandonExisting(string name, int waitType)
{
    using (Mutex m = new Mutex(false, name))
    {
        Task.Factory.StartNew(() => Assert.True(m.WaitOne(FailedWaitTimeout),
            CancellationToken.None, TaskCreationOptions.LongRunning, TaskScheduler.Default).Wait();
        switch (waitType)
        {
            case 0:
                Assert.Throws<AbandonedMutexException>(() => m.WaitOne(FailedWaitTimeout));
                break;
            case 1:
                AbandonedMutexException ame = Assert.Throws<AbandonedMutexException>(() => WaitHandle.WaitAny(new[] { m }, FailedWaitTimeout));
                Assert.Equal(0, ame.MutexIndex);
                Assert.Equal(m, ame.Mutex);
                break;
        }
    }
}
```

**Comment by stephentoub** [comment]
Thanks for updating these (and for the Mutex implementation!)

One comment, otherwise LGTM.

We should also look at adding a cross-process test of the naming, similar in concept to https://github.com/dotnet/corefx/blob/master/src/System.Threading/tests/SemaphoreTests.cs#L281-L317 .  That could be done as a separate PR if you wanted, though it might be useful here as well to help validate the CoreCLR change.

**Comment by kouvel** [comment]
I'm doing all of the cross-process testing in the PAL tests, and leaving the CoreFX tests to just do basic API-level testing (to cover the layer from the .NET API to the PAL Windows API). I'll take a closer look at the test you pointed to, and will see if I can add something similar.

**Comment by kouvel** [comment]
Looked at the test, I have tests like that in PAL tests already

File: src/System.Threading/tests/MutexTests.cs (hunk 1 above)
Comment (kouvel) [inline src/System.Threading/tests/MutexTests.cs]: Done

**Comment by kouvel** [comment]
@dotnet-bot test this please

**Comment by kouvel** [comment]
@dotnet-bot test this please

**Comment by kouvel** [comment]
@dotnet-bot test this please

**Comment by kouvel** [comment]
@dotnet-bot test this please

**Pull request merged**